src/main/app/case/to-api-format.ts

```diff
@@ -34,23 +34,19 @@ const fields = {
     D8MarriageDate: toApiDate(data.relationshipDate),
   }),
   helpWithFeesRefNo: (data: Case) => ({
-    D8HelpWithFeesReferenceNumber:
-      data.helpPayingNeeded === YesOrNo.Yes &&
-      data.alreadyAppliedForHelpPaying === YesOrNo.Yes &&
-      !isInvalidHelpWithFeesRef(data.helpWithFeesRefNo)
-        ? data.helpWithFeesRefNo
-        : '',
+    D8HelpWithFeesReferenceNumber: !isInvalidHelpWithFeesRef(data.helpWithFeesRefNo) ? data.helpWithFeesRefNo : '',
   }),
   englishOrWelsh: (data: Case) => ({
     LanguagePreferenceWelsh: data.englishOrWelsh === LanguagePreference.Welsh ? YesOrNo.Yes : YesOrNo.No,
   }),
 };
 
 const toApiDate = (date: CaseDate | undefined) => {
-  if (!date?.year || !date?.month || !date?.day) {
+  const dateParts = ['year', 'month', 'day'];
+  if (!date || !dateParts.every(r => Object.keys(date).includes(r))) {
```

Comment:
imho it was more readable before but I'm not too fussed

Reply:
👍 I've swapped this back in a6e5eca.